util_test.go

```diff
@@ -44,6 +47,34 @@ func TestSimpleTokenAccessor(t *testing.T) {
 	}
 }
 
+func TestSimpleTokenAccessorGetTokensSynchronization(t *testing.T) {
+	accessor := getSimpleTokenAccessor()
+	var wg sync.WaitGroup
+	failed := false
+	for i := 0; i < 1000; i++ {
+		wg.Add(1)
+		go func() {
+			// set a random session and token
+			session := rand.Int()
+			sessionStr := strconv.Itoa(session)
+			accessor.SetTokens("t"+sessionStr, "m"+sessionStr, session)
+
+			// read back session and token and verify that invariant still holds
+			token, masterToken, session := accessor.GetTokens()
+			sessionStr = strconv.Itoa(session)
+			if "t"+sessionStr != token || "m"+sessionStr != masterToken {
```

Comment:
This is only testing that the master token and session token are set atomically, which is an implementation detail of a particular accessor. I think a more robust test would be to ensure that only a single session renewal occurs when multiple goroutines encounter expired sessions, but I don't know how hard that would be to mock out.

Reply:
The other unit test, `TestUnitTokenAccessorRenewSession` is calling with multiple goroutines competing to issue renewal, and verifying at the end that only 1 renewal actually happened.